src/eolab/rastertools/cli/hillshade.py

```diff
@@ -71,6 +71,7 @@ def create_argparser(rastertools_parsers):
              "that lists the input files to process (one input file per line in .lst)")
     cli.with_outputdir_arguments(parser)
     cli.with_window_arguments(parser)
+    cli.with_max_radius_arguments(parser)
```

Comment:
I think it would be better to add max_radius as a specific argument, as it is done for radius, see on line 46.

Reply:
Ok I reworked max_radius as an Hillshade class' attribute

Reply:
Indeed I can use the `self.radius` as a `self.max_radius` threshold value in Hillshade in the following way below (`rastertools/hillshade.py`)
      
```
        wmax = None
        wmin = None
        with rasterio.open(inputfile) as src:
            if src.count != 1:
                raise ValueError("Invalid input file, it must contain a single band.")
            for ji, window in src.block_windows(1):
                data = src.read(1, masked=True, window=window)
                wmax = max(wmax, float(data.max())) if wmax is not None else float(data.max())
                wmin = min(wmin, float(data.min())) if wmin is not None else float(data.min())

        delta = int((wmax - wmin) / self.resolution)
        radius = int(delta / np.tan(np.radians(self.elevation)))

        if self.max_radius is not None and radius > self.max_radius:
            _logger.info(f"The radius value is {radius} exceeding {self.max_radius} threshold. "
                         f"Radius is reset to {self.max_radius}.")
            radius = self.max_radius

```
It looks smoother than my previous commit

Reply:
Maybe we can change the logic in `hillshade.py` without adding a new parameter. Let say `radius` and `max_radius` are the same parameter `radius`. In `hillshade.py`, we can systematically compute the optimal one (I name it `optimal_radius` in the following). 

1. If `optimal_radius` is lower than `radius`, we use `optimal_radius` since this is useless to explore pixels further than `radius`. 
2. If `optimal_radius` is higher than `radius`, we use `radius` and can send a warning message saying that the result may miss shadow pixels.

Reply:
This sounds perfect for our needs, so if it is OK for you, let's do this!